src/extension/src/TelemetryWriter.py

```diff
@@ -13,75 +13,107 @@
+    """Class for writing telemetry data to events"""
 
     def __init__(self):
-        self.data_transports = []
-        self.activity_id = None
-
-        # Init state report
-        self.send_ext_state_info('Started Linux patch extension execution.')
-        self.send_machine_config_info()
-
-    # region Primary payloads
-    def send_ext_state_info(self, state_info):
-        # Expected to send up only pivotal extension state changes
-        return self.try_send_message(state_info, Constants.TelemetryExtState)
-
-    def send_config_info(self, config_info, config_type='unknown'):
-        # Configuration info
-        payload_json = {
-            'config_type': config_type,
-            'config_value': config_info
-        }
-        return self.try_send_message(payload_json, Constants.TelemetryConfig)
-
-    def send_error_info(self, error_info):
-        # Expected to log significant errors or exceptions
-        return self.try_send_message(error_info, Constants.TelemetryError)
-
-    def send_debug_info(self, error_info):
-        # Usually expected to instrument possibly problematic code
-        return self.try_send_message(error_info, Constants.TelemetryDebug)
-
-    def send_info(self, info):
-        # Usually expected to be significant runbook output
-        return self.try_send_message(info, Constants.TelemetryInfo)
-    # endregion
-
-    # Composed payload
-    def send_machine_config_info(self):
-        # Machine info
-        machine_info = {
-            # The commented out code needs to be addressed similar to how Core does it for Python 3.8+ support.
-            # Not an issue at the time of this writing as telemetry support has not been implemented.
-            #  'platform_name': str(platform.linux_distribution()[0]),
-            #  'platform_version': str(platform.linux_distribution()[1]),
-            'machine_arch': str(platform.machine())
-        }
-        return self.send_config_info(machine_info, 'machine_config')
-
-    def send_execution_error(self, cmd, code, output):
-        # Expected to log any errors from a cmd execution, including package manager execution errors
-        error_payload = {
-            'cmd': str(cmd),
-            'code': str(code),
-            'output': str(output)[0:3072]
+        self.events_folder_path = None
+        self.operation_id = ""
+
+    def __new_event_json(self, task_name, event_level, message):
+        return {
+            "Version": Constants.EXT_VERSION,
+            "Timestamp": str((datetime.datetime.utcnow()).strftime("%Y-%m-%dT%H:%M:%SZ")),
+            "TaskName": task_name,
+            "EventLevel": event_level,
+            "Message": self.__ensure_message_restriction_compliance(message),
+            "EventPid": "",
+            "EventTid": "",
+            "OperationId": self.operation_id  # we can provide activity id from config settings here, but currently we only read settings file for enable command
```

Comment:
It's good to have comments like this. Helps with long term maintainability.

So long as we know what non-enable command is running and the time actions related to it occurred, that's a good start.